python/bootstrap.py

```diff
@@ -383,6 +383,69 @@ def setup_projects(backend):
+  if status == False:
+    exit(1)
+    
+  files = [f for f in listdir("./autocorrect_config") if isfile(join("./autocorrect_config", f)) and f.endswith("_field_type.json")]
+  for file in files:
+    print ("Creating typeahead field_type for %s" % file)
+    backend.add_field_type(collection_id, json.load(open(join("./autocorrect_config", file))))
+
+  pipe_files = [f for f in listdir("./autocorrect_config") if isfile(join("./autocorrect_config", f)) and f.endswith("_pipeline.json")]
+  for file in pipe_files:
+    print ("Creating Pipeline for %s" % file)
+    if file.find("query") != -1:
+      backend.create_pipeline(json.load(open(join("./autocorrect_config", file))), pipe_type="query-pipelines")
+    else:
+      backend.create_pipeline(json.load(open(join("./autocorrect_config", file))))
+
+  # set the fields to ngram_query and edge_ngram_query to enable autocorrect. It breaks down the query before matching.
+  print ("Creating fields")
+  backend.add_field(collection_id, "name_contains", type="ngram_query", stored="true", multivalued="false")
+  backend.add_field(collection_id, "name_edge", type="edge_ngram_query", stored="true", multivalued="false")
+  backend.add_field(collection_id, "name_en", type="text_en", stored="true", multivalued="false")
+  backend.add_field(collection_id, "name_no_vowels", type="text_no_vowels", stored="true", multivalued="false")
+  backend.add_field(collection_id, "name_phonetic_en", type="phonetic_en", stored="true", multivalued="false")
+  backend.add_field(collection_id, "name_sort", type="string_sort", stored="false", multivalued="false")
+  backend.add_field(collection_id, "spell", type="text_general", stored="false", multivalued="false")
+  
+  backend.add_field(collection_id, "name", type="text_general", multivalued="false", stored="true", copyDests=["name_edge", "name_contains", "name_no_vowels", "name_phonetic_en", "name_en", "name_sort", "spell"])
+  backend.add_field(collection_id, "type", type="string", stored="true")
+  backend.add_field(collection_id, "synonyms", type="text_general", stored="true", multivalued="true")
+  backend.add_field(collection_id, "bh_search_score", type="int", stored="true")
+  backend.add_field(collection_id, "bh_rank", type="int", stored="true")
+  print ("Finished creating fields")
+  
+  # change of the path where files are located. i.e. ./autocorrect_config
+  print ("Creating datasource")
+  datasource_files = [f for f in listdir("./autocorrect_config") if isfile(join("./autocorrect_config", f)) and f.endswith("_datasource.json")]
+  fusion_update_url = app.config['FUSION_URLS'][0] + "apollo/connectors/datasources"
+  FUSION_USERNAME = app.config.get("FUSION_ADMIN_USERNAME", "admin")
+  FUSION_PASSWORD = app.config.get("FUSION_ADMIN_PASSWORD")
+  for file in datasource_files:
+    resp = requests.post(fusion_update_url,
+                                 data=json.dumps(json.load(open(join("./autocorrect_config", file)))),
```

Comment:
The datasource is the same; `lucidfind`collection. But the fieldtype of `name_contains` and `name_edge` in my collection is different. That's why I create a separate collection.

Actually we can keep only one collection --activate_autocorrect_typeahead will create a collection which will provide both typeahead and autocorrect features